src/main/resources/jupyter/init-actions.sh

```diff
@@ -4,37 +4,38 @@ set -e -x
 
 # adapted from https://github.com/GoogleCloudPlatform/dataproc-initialization-actions/blob/master/datalab/datalab.sh
 
-function update_apt_get() {
-  for ((i = 0; i < 10; i++)); do
-    if apt-get update; then
+# Retry a command up to a specific number of times until it exits successfully,
+# with exponential back off.
+#
+# $ retry 5 echo "Hello"
+# Hello
+#
+# $ retry 5 false
+# Retry 1/5 exited 1, retrying in 2 seconds...
+# Retry 2/5 exited 1, retrying in 4 seconds...
+# Retry 3/5 exited 1, retrying in 8 seconds...
+# Retry 4/5 exited 1, retrying in 16 seconds...
+# Retry 5/5 exited 1, no more retries left.
+#
+function retry {
+  local retries=$1
+  shift
+
+  local count=0
```

Comment:
is `count` used? I don't find usage in `retry` function.

Reply:
Oops, leftover from when I used `while` instead of `for`